packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts

```diff
@@ -5,55 +5,90 @@
 
 import * as vscode from 'vscode'
 import * as fs from 'fs'
+import * as path from 'path'
 import { getLogger, Logger } from '../../shared/logger'
 import { localize } from '../../shared/utilities/vsCodeUtils'
 import { VueWebview } from '../../webviews/main'
 import { ExtContext } from '../../shared/extensions'
-//import { telemetry } from '../../shared/telemetry/telemetry'
+// TODO: Implement Telemetry ... import { telemetry } from '../../shared/telemetry/telemetry'
 import { AccessAnalyzer, S3 } from 'aws-sdk'
-import { exec } from 'child_process'
+import { execSync } from 'child_process'
+import { ToolkitError } from '../../shared/errors'
 const AmazonS3URI = require('amazon-s3-uri')
 
+const defaultTerraformConfigPath = path.join(__dirname, '..', 'terraformConfig', 'default.yaml')
+const tempFilePath = path.join(__dirname, 'policyDocument.json')
+const customPolicyCheckDiagnosticCollection = vscode.languages.createDiagnosticCollection('customPolicyCheck')
+const validatePolicyDiagnosticCollection = vscode.languages.createDiagnosticCollection('validatePolicy')
+
 export interface IamPolicyChecksInitialData {
     referenceFilePath: string
-    cfnParameterPath: string
     referenceDocument: string
+    referenceFileErrorMessage: string
+    cfnParameterPath: string
     pythonToolsInstalled: boolean
 }
 
 export class IamPolicyChecksWebview extends VueWebview {
     public static readonly sourcePath: string = 'src/accessanalyzer/vue/index.js'
     public readonly id = 'iamPolicyChecks'
-
-    //private readonly logger = getLogger()
+    private static editedDocumentUri: vscode.Uri = vscode.window.activeTextEditor?.document.uri!
+    private static editedDocumentFileName: string = vscode.window.activeTextEditor?.document.uri.path!
+    private static editedDocument: string = ''
 
     public constructor(
         private readonly data: IamPolicyChecksInitialData,
         private readonly client: AccessAnalyzer,
+        private readonly region: string,
```

Comment:
Q: does the region here has to match with the region used in the credentials?

Reply:
Can you confirm this? I think this should be the destination region where the resources are being deployed to -https://github.com/awslabs/aws-cloudformation-iam-policy-validator. I think it may give incorrect results if we fetch region from the AWS credentials when they are trying to deploy to a different region